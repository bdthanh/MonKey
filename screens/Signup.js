import React, {useState, useEffect} from 'react'
import MainContainer from '../components/Containers/Main';
import KeyboardAvoidingContainer from '../components/Containers/KeyboardAvoiding';
import { Text, View, Alert } from 'react-native';
import styles from '../components/styles';
import TextInputWithIcon from '../components/Containers/TextInputWithIcon';
import SignUpButton from '../components/Containers/SignUpButton';
import PressableText from '../components/Containers/PressableText';
import { handleSignup } from '../utils/authentication';
import { StatusBar } from 'expo-status-bar';
import Login from './Login';

const SignUp = ({navigation}) => {
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [cfPassword, setCfPassword] = useState("");
  const handleSignUp1 = (username, email, pass, confirmPass) => {
    if (username != '' && email != '' && pass == confirmPass) {
      handleSignup(username, email, pass);
      navigation.navigate('Login')
    } else if (username == '') {
      Alert.alert("Alert", "Enter your username", [
        {text: 'Understand', onPress: () => console.log('Alert closed')}
      ]);
    } else if (email == '') {
      Alert.alert("Alert", "Enter your email", [
        {text: 'Understand', onPress: () => console.log('Alert closed')}
      ]);
    } else {
      Alert.alert("Alert", "Check your confirm password", [
        {text: 'Understand', onPress: () => console.log('Alert closed')}
      ]);
    }
  }
  return (
    <>
      <MainContainer>
        <KeyboardAvoidingContainer>
          <View style={{alignContent: 'flex-start', paddingBottom: 20}}>
            <Text style={styles.boldBlueText}>Sign-up</Text>
          </View>
            <TextInputWithIcon
              label='Username'
              icon='user'
              placeholder='YourUsername'
              value={username}
              autoCapitalize = 'none'
              onChangeText={text => setUsername(text)}
            />
            <TextInputWithIcon
              label='Email address'
              icon='mail'
              placeholder='yourmail@gmail.com'
              keyboardType='email-address'
              value={email}
              autoCapitalize = 'none'
              onChangeText={text => setEmail(text)}
            />
            <TextInputWithIcon
              label='Password'
              icon='lock-open'
              placeholder='* * * * * * * *'
              isPassword={true}
              value={password}
              autoCapitalize = 'none'
              onChangeText={text => setPassword(text)}
            />
            <TextInputWithIcon
              label='Confirm password'
              icon='lock-open'
              placeholder='* * * * * * * *'
              isPassword={true}
              value={cfPassword}
              autoCapitalize = 'none'
              onChangeText={text => setCfPassword(text)}
            />
            <SignUpButton onPress={() => {handleSignUp1(username, email, password, cfPassword)}}>
              Sign-up
            </SignUpButton>
            <PressableText onPress={() => navigation.navigate('Login')}>
              Have an account? Login here
            </PressableText>
            <PressableText onPress={() => navigation.navigate('ForgotPassword')}>
              Forgot password?
            </PressableText>
        </KeyboardAvoidingContainer>
      </MainContainer>
    </>
  )
}

export default SignUp;