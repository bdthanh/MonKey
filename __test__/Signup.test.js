import React from 'react';
import {create} from 'react-test-renderer'
import Signup from '../screens/Signup';

const tree = create(<Signup/>);

test('snapshot Signup', () => {
  expect(tree).toMatchSnapshot();
});
