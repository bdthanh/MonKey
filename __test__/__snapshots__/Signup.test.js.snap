// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFAEB",
      "flex": 1,
      "padding": 25,
      "paddingTop": 50,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignContent": "flex-start",
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 42,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Sign-up
            </Text>
          </View>
          <View>
            <View
              style={
                Object {
                  "borderRightColor": "#35baf6",
                  "borderRightWidth": 2,
                  "left": 13,
                  "paddingRight": 8,
                  "position": "absolute",
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Username
            </Text>
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="YourUsername"
              placeholderTextColor="#ADD8E6"
              style={
                Object {
                  "backgroundColor": "#FFFAEB",
                  "borderColor": "#35baf6",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#35baf6",
                  "fontSize": 18,
                  "height": 60,
                  "marginBottom": 25,
                  "marginTop": 3,
                  "padding": 15,
                  "paddingLeft": 60,
                  "paddingRight": 55,
                }
              }
              value=""
            />
          </View>
          <View>
            <View
              style={
                Object {
                  "borderRightColor": "#35baf6",
                  "borderRightWidth": 2,
                  "left": 13,
                  "paddingRight": 8,
                  "position": "absolute",
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Email address
            </Text>
            <TextInput
              autoCapitalize="none"
              keyboardType="email-address"
              onChangeText={[Function]}
              placeholder="yourmail@gmail.com"
              placeholderTextColor="#ADD8E6"
              style={
                Object {
                  "backgroundColor": "#FFFAEB",
                  "borderColor": "#35baf6",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#35baf6",
                  "fontSize": 18,
                  "height": 60,
                  "marginBottom": 25,
                  "marginTop": 3,
                  "padding": 15,
                  "paddingLeft": 60,
                  "paddingRight": 55,
                }
              }
              value=""
            />
          </View>
          <View>
            <View
              style={
                Object {
                  "borderRightColor": "#35baf6",
                  "borderRightWidth": 2,
                  "left": 13,
                  "paddingRight": 8,
                  "position": "absolute",
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Password
            </Text>
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="* * * * * * * *"
              placeholderTextColor="#ADD8E6"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#FFFAEB",
                  "borderColor": "#35baf6",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#35baf6",
                  "fontSize": 18,
                  "height": 60,
                  "marginBottom": 25,
                  "marginTop": 3,
                  "padding": 15,
                  "paddingLeft": 60,
                  "paddingRight": 55,
                }
              }
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 13,
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "borderRightColor": "#35baf6",
                  "borderRightWidth": 2,
                  "left": 13,
                  "paddingRight": 8,
                  "position": "absolute",
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Confirm password
            </Text>
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              placeholder="* * * * * * * *"
              placeholderTextColor="#ADD8E6"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#FFFAEB",
                  "borderColor": "#35baf6",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#35baf6",
                  "fontSize": 18,
                  "height": 60,
                  "marginBottom": 25,
                  "marginTop": 3,
                  "padding": 15,
                  "paddingLeft": 60,
                  "paddingRight": 55,
                }
              }
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 13,
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#35baf6",
                "borderRadius": 10,
                "flex": 1,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFAEB",
                  "fontSize": 21,
                  "fontWeight": "500",
                }
              }
            >
              Sign-up
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Have an account? Login here
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Forgot password?
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
